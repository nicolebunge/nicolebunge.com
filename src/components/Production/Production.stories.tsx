import Production from '.';

export default {
  component: Production,
};

export const Default = {
  args: {
    name: 'Erbarmen',
    role: 'Lis SÃ¸rensen',
    organization: {
      name: 'Berliner Kriminal Theater',
      slug: 'berliner-kriminal-theater',
      url: 'https://www.kriminaltheater.de',
    },
    directors: [
      {
        name: 'Wolfgang Rumpf',
      },
    ],
  },
};
