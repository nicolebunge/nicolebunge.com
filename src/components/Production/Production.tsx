import React from 'react';
import Link from '../Link';

function formatList(list: string[]): string {
  return list.reduce(
    (text, value, i, array) => text + (i < array.length - 1 ? ', ' : ' & ') + value,
  );
}

interface Director {
  name: string;
  url?: string;
}

interface Organization {
  fields: {
    slug: string;
  };
  name: string;
  url: string;
}

export interface ProductionProps {
  directors: Director[];
  // eslint-disable-next-line react/no-unused-prop-types
  id: string;
  name: string;
  organization: Organization;
  role: string;
}

function Production(props: ProductionProps): JSX.Element {
  const { name, role, directors, organization } = props;

  return (
    <p>
      <strong>„{name}“</strong>, {role}
      <br />
      Regie: {formatList(directors.map((director) => director.name))},{' '}
      <Link to={organization.url}>{organization.name}</Link>
    </p>
  );
}

export default Production;
