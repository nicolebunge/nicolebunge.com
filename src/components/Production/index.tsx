import Production from './Production';

export * from './Production';
export default Production;
