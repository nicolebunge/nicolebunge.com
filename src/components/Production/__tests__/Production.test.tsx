import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';
import * as stories from '../Production.stories';

const { Default } = composeStories(stories);

describe('Production component', () => {
  it('renders correctly', async () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
