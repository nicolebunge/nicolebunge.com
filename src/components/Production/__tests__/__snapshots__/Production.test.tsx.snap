// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Production component > renders correctly 1`] = `
<p>
  <strong>
    „
    Erbarmen
    “
  </strong>
  , 
  Lis Sørensen
  <br />
  Regie: 
  Wolfgang Rumpf
  ,
   
  <a
    href="https://www.kriminaltheater.de"
    rel="noopener noreferrer"
    target="_blank"
  >
    Berliner Kriminal Theater
  </a>
</p>
`;
